File: server/src/models/notification.rs
use chrono::{DateTime, Utc};
use serde::Serialize;

/// 알림 유형
#[derive(Debug, Clone, PartialEq, Eq, Serialize, sqlx::Type)]
#[sqlx(type_name = "TEXT", rename_all = "snake_case")]
pub enum NotificationType {
    PriceAlert,
    CategoryAlert,
    KeywordAlert,
    Referral,
    Event,
    System,
}

/// notifications 테이블
#[derive(Debug, Clone, sqlx::FromRow, Serialize)]
pub struct Notification {
    pub id: i64,
    pub user_id: i64,
    pub notification_type: NotificationType,
    pub reference_id: Option<i64>,
    pub reference_type: Option<String>,
    pub title: String,
    pub body: Option<String>,
    pub deep_link: Option<String>,
    pub is_read: bool,
    pub sent_at: DateTime<Utc>,
    pub read_at: Option<DateTime<Utc>>,
}

File: app/lib/models/notification.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'notification.freezed.dart';
part 'notification.g.dart';

/// 알림 내역
@freezed
abstract class AppNotification with _$AppNotification {
  const factory AppNotification({
    required int id,
    @JsonKey(name: 'user_id') required int userId,
    @JsonKey(name: 'notification_type') required String notificationType,
    @JsonKey(name: 'reference_id') int? referenceId,
    @JsonKey(name: 'reference_type') String? referenceType,
    required String title,
    String? body,
    @JsonKey(name: 'deep_link') String? deepLink,
    @JsonKey(name: 'is_read') @Default(false) bool isRead,
    @JsonKey(name: 'sent_at') DateTime? sentAt,
    @JsonKey(name: 'read_at') DateTime? readAt,
  }) = _AppNotification;

  factory AppNotification.fromJson(Map<String, dynamic> json) =>
      _$AppNotificationFromJson(json);
}
